config/config.go

```diff
@@ -216,123 +215,20 @@ func NewDefaultAgentConfig() *AgentConfig {
 
 // NewAgentConfig returns an AgentConfig using a configuration file. It can be nil
 // if there is no file available. In this case we'll configure only via environment.
-func NewAgentConfig(agentIni *File, yamlPath, netYamlPath string) (*AgentConfig, error) {
+func NewAgentConfig(iniPath, yamlPath, netYamlPath string) (*AgentConfig, error) {
 	var err error
 	cfg := NewDefaultAgentConfig()
 
-	var ns string
-	var section *ini.Section
-	if agentIni != nil {
-		section, _ = agentIni.GetSection("Main")
-	}
-
-	// Pull from the ini Agent config by default.
-	if section != nil {
-		a, err := agentIni.Get("Main", "api_key")
-		if err != nil {
+	if util.PathExists(yamlPath) { // For Agents >= 6 we will have a YAML config file to use.
```

Comment:
Should we log something if both path exists ? (Since it might be breaking for some people)

Reply:
Good Idea. I can do this and say we're preferring YAML